Source/JavaScriptCore/bytecode/Repatch.cpp

```diff
@@ -665,6 +665,38 @@ static InlineCacheAction tryCacheArrayGetByVal(JSGlobalObject* globalObject, Cod
             case ArrayStorageShape:
                 accessType = AccessCase::IndexedArrayStorageLoad;
                 break;
+            case NoIndexingShape: {
+                if (!base->isObject())
+                    return GiveUpOnCache;
+
+                if (base->structure()->mayInterceptIndexedAccesses() || base->structure()->typeInfo().interceptsGetOwnPropertySlotByIndexEvenWhenLengthIsNotZero())
+                    return GiveUpOnCache;
```

Comment:
won't this be handled in the `generateConditionsForIndexedMiss` call?

Reply:
`generateConditions...` functions only checks Structures of prototype or upper. So for instance's Structure, we need to check it here before calling it.

Reply:
right, makes sense